### ISSUE TITLE
Documentation Change - Firefox Extension in iOS on Orion

### ISSUE THREAD

**Issue by rickdekker** [Lartsch/FediAct#16]
It is now possible to use Firefox extensions on iOS using Orion: https://apps.apple.com/gb/app/orion-browser-by-kagi/id1484498200

**Comment by Lartsch** [comment]
Awesome, thank you for letting know! Added to the readme.

**Comment by ricklamers** [comment]
@rickdekker did you actually get this to work? Via “Install from File” I get a no-op. Tried multiple versions of the zip file of the firefox extension.

**Comment by ricklamers** [comment]
I was able to install by navigating to https://addons.mozilla.org/af/firefox/addon/fediact/ and clicking in “Download file”. That prompted install and seemed to work.

Not sure if the extension itself is working, after configuring an instance URL in the settings nothing seems to happen on follow actions on Mastodon profile pages of other instances.

**Comment by stefandesu** [comment]
Doesn't seem to work for me either. I'm using Orion for Mac though. I tried basically everything: Chrome extension from store, Firefox extension from store, Firefox extension from GitHub Releases. The icon shows up and seems like everything's fine, but it's not working. Works really well in Firefox though.

**Comment by Lartsch** [comment]
Okay, I will add to the readme that this is an untested solution / not guaranteed to work. I can't test / debug it myself, since I own no Apple devices. If anyone wants to make the effort to debug it, feel free and let me know if you need help.

**Comment by Lartsch** [comment]
The first important information would be:
- Does the popup open at all when the icon is clicked?
- Are settings actually saved when submitting and re-opening the popup?
- Are there any errors thrown by the background or content script?

**Comment by stefandesu** [comment]
The popup works and it does save the information. I can't see any extension-related errors on the console. If I can find the time, I'll try to do more debugging on this!

It might not actually be an issue with the extension itself. It is known that some extensions just don't work yet with Orion and it is on them to fix it. I might open a thread on their feedback forum about it.

**Comment by stefandesu** [comment]
I did a bit more debugging (although I have no experience with making browser extensions, so no idea what I'm doing) and summarized my findings in a post on the Orion feedback forums: https://orionfeedback.org/d/3946-extension-fediact-not-working

Seems to be two separate issues for the Firefox vs. Chrome extensions, and I'm hoping the people on the forums can provide help why it's not working.

**Comment by stefandesu** [comment]
Okay, I tried to look into it a bit more, using the source and modifying files. Still no idea about that Firefox error. With the Chrome extension however, it seems like they don't support the `chrome` namespace **in background service worker**. During injection, those variables are available like you'd expect. I feel like they understood the issue (they already added the "Planned" tag to my thread) and will probably fix it at some point.

I guess it would be possible, in theory, to do everything during injection time, right? Or will there be issues with CORS when communicating with the home instance? Even if it was possible, it's probably not worth investing much time into it if the fix on their side is rather trivial.

**Comment by Lartsch** [comment]
> I guess it would be possible, in theory, to do everything during injection time, right?

No, sadly not. There are cases were the background script is absolute required, for example:
- Grabbing the token, since requests made in the background script use existing cookies for domains
- Resolving by 302 redirects, since only requests in the background script can access the HTTP Location header

I also think we should let them do their work. If the code base supports Firefox & Chromium, I'm sure they can figure it out.
